### DISCUSSION TITLE
How to detect a Lemmy instance page ?

### DISCUSSION THREAD

**Discussion by KaKi87** [LemmyNet/lemmy-ui#1927] [Q&A] [answered]
Hey there,

I was using the `lemmyConfig` global variable for my [Fediverse redirector](https://git.kaki87.net/KaKi87/userscripts/src/branch/master/fediverseRedirector/README.md) userscript to detect that a web page is a Lemmy instance.

If that variable gets removed with #1907 then what could I use now ?

I've looked at the HTML elements multipe times and found nothing promising.

Thanks

**Comment by Zetaphor** [comment, marked as answer]
Assuming the server is running at least 0.18.1, there is a `lemmy-site` class on `#app`

![image](https://github.com/LemmyNet/lemmy-ui/assets/3112763/643081ae-b069-42a5-bb58-341f386f8352)

**Reply by KaKi87** [reply]
I didn't imagine that they'd add this at the same time as removing the variable, that's a nice coincidence. Thanks

**Reply by Zetaphor** [reply]
This was added along with a number of other CSS additions in 0.18.1. See #1327

**Reply by KaKi87** [reply]
Wow, it's nice to see you've taken userscripts into consideration. :)

**Reply by Zetaphor** [reply]
Do keep in mind that these are subject to change. Support for custom themes and userscripts is provided as-is, and could break without notice. The core maintainers want to prevent creating new tickets around dependency issues and backwards compatibility.

**Reply by KaKi87** [reply]
I understand, although I also hope this class will remain there.

When creating a userscript, I know I have to make do.

But working on self-hostable (and usually open) apps is different than working with centralized (and usually proprietary) ones&nbsp;: while the latter requires dealing with higher unpredictability and lower accessibility, the former requires dealing with non URL-based detection.
